src/OperatorEnumConstruction.jl

```diff
@@ -107,12 +108,17 @@ end
     return mapping[f]
 end
 
-function empty_all_globals!()
-    LATEST_OPERATORS.x = nothing
-    LATEST_OPERATORS_TYPE.x = IsNothing
-    empty!(LATEST_UNARY_OPERATOR_MAPPING)
-    empty!(LATEST_BINARY_OPERATOR_MAPPING)
-    LATEST_VARIABLE_NAMES.x = String[]
+function empty_all_globals!(; force=true)
+    if force || islocked(LATEST_LOCK)
+        lock(LATEST_LOCK) do
+            LATEST_OPERATORS.x = nothing
+            LATEST_OPERATORS_TYPE.x = IsNothing
+            empty!(LATEST_UNARY_OPERATOR_MAPPING)
+            empty!(LATEST_BINARY_OPERATOR_MAPPING)
+            LATEST_VARIABLE_NAMES.x = String[]
+            unlock(LATEST_LOCK)
+        end
+    end
     return nothing
```

Comment:
Might want to produce a warning here if locked? I can't think of a case where you would want to do this. If a user wants to legitimately do this from multiple processes it's likely better they just use `parse_expression` instead!